spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/AbstractConnectionFactory.java

```diff
@@ -525,17 +525,20 @@ protected String getBeanName() {
 
 	/**
 	 * When {@link #setAddresses(String) addresses} are provided and there is more than
-	 * one, set to true to shuffle the list before opening a new connection so that the
-	 * connection to the broker will be attempted in random order.
+	 * one, set to true (default) to shuffle the list before opening a new connection so
+	 * that the connection to the broker will be attempted in random order.
 	 * @param shuffleAddresses true to shuffle the list.
 	 * @since 2.1.8
-	 * @deprecated since 2.3 in favor of
+	 * @deprecated since 2.3 in favor of {@link #setAddressShuffleMode(AddressShuffleMode)}.
 	 * @see Collections#shuffle(List)
-	 * {@link #setAddressShuffleMode(AddressShuffleMode)}.
 	 */
 	@Deprecated
 	public void setShuffleAddresses(boolean shuffleAddresses) {
-		setAddressShuffleMode(AddressShuffleMode.RANDOM);
+		if (shuffleAddresses) {
+			setAddressShuffleMode(AddressShuffleMode.RANDOM);
+		} else {
+			setAddressShuffleMode(AddressShuffleMode.NONE);
+		}
```

Comment:
This deprecated method can/should be removed in 3.0.

Reply:
Done